### DISCUSSION TITLE
Http response codes (no helper class)

### DISCUSSION THREAD

**Discussion by yildirimmurat** [yiisoft/yii2#19864] [Q&A] [answered]
Hi there,

I am new to this framework so, this question might be trivial. I am trying to set statusCode on my rest api and lets say for 401 code i need to write like 

`                \Yii::$app->response->setStatusCode(401);
`

i was checking if there is a helper class and so that i can get some constants for http response codes like 

`                \Yii::$app->response->setStatusCode(Response::UNAUTHORIZED);
`

I do not want to write any helper class for this myself so I checked if Yii2 has any helper for this.. There is none if I am not mistaken. I do not want to do something like this because its just overkill:

`                \Yii::$app->response->setStatusCode((new UnauthorizedHttpException())->statusCode);
`

So, for now I am using symfonys helper class:

`                \Yii::$app->response->setStatusCode(\Symfony\Component\HttpFoundation\Response::HTTP_UNAUTHORIZED);
`

I just wanted to state that would be nice if Yii2 has something similar..

Best regards

**Comment by bizley** [comment, marked as answer]
We are not planning to add constants for that. List of statuses is available at `yii\web\Response::$httpStatuses`.

**Comment by yildirimmurat** [comment]
Thanks for quick reply @bizley . The aim is actually writing a more readable code, one can find codes easily. But, allright I noted that its not on future plans

**Comment by tomaszkane** [comment]
Why not use helper from Yii3?
https://github.com/yiisoft/http/#http-status-codes

**Reply by samdark** [reply]
Absolutely could be used same as many Yii3 components already released.
